tests/rptest/tests/timequery_test.py

```diff
@@ -623,6 +625,58 @@ def test_timequery_with_spillover_gc_delayed(self):
                 else:
                     assert offset == o, f"Expected {o}, got {offset}"
 
+    @cluster(num_nodes=4)
+    def test_timequery_empty_local_log(self):
+        self.set_up_cluster(cloud_storage=True,
+                            batch_cache=False,
+                            spillover=False)
+
+        total_segments = 3
+        record_size = 1024
+        base_ts = 1664453149000
+        msg_count = (self.log_segment_size * total_segments) // record_size
+        local_retention = 1  # Any value works for this test.
+        topic, timestamps = self._create_and_produce(self.redpanda, True,
+                                                     local_retention, base_ts,
+                                                     record_size, msg_count)
+
+        # Confirm messages written
+        rpk = RpkTool(self.redpanda)
+        p = next(rpk.describe_topic(topic.name))
+        assert p.high_watermark == msg_count
+
+        # Restart the cluster to force segment roll. The newly created segment
+        # will have no user data which is what we want to test.
+        self.redpanda.restart_nodes(self.redpanda.nodes)
+        wait_until(lambda: len(list(rpk.describe_topic(topic.name))) > 0,
+                   30,
```

Comment:
nit: mentioned in slack, you can also trigger leadership changes and that should also force a roll